Source/JavaScriptCore/runtime/JSIteratorPrototype.cpp

```diff
@@ -123,4 +128,28 @@ JSC_DEFINE_CUSTOM_SETTER(iteratorProtoToStringTagSetter, (JSGlobalObject* global
     return JSValue::encode(jsUndefined());
 }
 
+// https://tc39.es/proposal-iterator-helpers/#sec-iteratorprototype.toarray
+JSC_DEFINE_HOST_FUNCTION(iteratorProtoFuncToArray, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSValue thisValue = callFrame->thisValue();
+    if (!thisValue.isObject())
+        return throwVMTypeError(globalObject, scope, "Iterator.prototype.toArray requires that |this| be an Object."_s);
+
+    MarkedArgumentBuffer value;
+    forEachInIteratorProtocol(globalObject, thisValue, [&value, &scope](VM&, JSGlobalObject* globalObject, JSValue nextItem) {
+        value.append(nextItem);
+        if (UNLIKELY(value.hasOverflowed()))
+            throwOutOfMemoryError(globalObject, scope);
+    });
```

Comment:
We should have a fast path before reaching to here.
When the input is fast array, it should be array cloning operation.

Reply:
How to clone an array? `JSArray::createWithButterfly` with the butterfly of the original array?

Reply:
OK, please put FIXME comment pointing at the bug for that fast implementation here.
I think we need to implement it before enabling this feature.